src/main/java/org/digitalstack/chess/ChessBoard.java

```diff
@@ -12,10 +12,24 @@ public ChessBoard() {
     }
 
     public void add(Pawn pawn, int xCoordinate, int yCoordinate, PieceColor pieceColor) {
-        throw new UnsupportedOperationException("Need to implement ChessBoard.add()");
+        if (isLegalBoardPosition(xCoordinate, yCoordinate) && pieces[xCoordinate][yCoordinate] == null) {
```

Comment:
Maybe swap the condition and fail fast. Otherwise it's hard to follow when the coordonates are set to -1.